tidy3d/components/simulation.py

```diff
@@ -824,6 +824,26 @@ def plot_grid(
         # Plot snapping points
         for point in self.grid_spec.snapping_points:
             _, (x_point, y_point) = Geometry.pop_axis(point, axis=axis)
+            if x_point is None and y_point is None:
+                continue
+            if x_point is None:
+                ax.axhline(
+                    y=self._evaluate_inf(y_point),
+                    xmin=0.1,
+                    xmax=0.9,
+                    linewidth=2 * kwargs["linewidth"],
+                    color=plot_params.edgecolor,
+                )
+                continue
+            if y_point is None:
+                ax.axvline(
+                    x=self._evaluate_inf(x_point),
+                    ymin=0.1,
+                    ymax=0.9,
+                    linewidth=2 * kwargs["linewidth"],
+                    color=plot_params.edgecolor,
+                )
+                continue
```

Comment:
@dmarek-flex @tylerflex here I enable `snapping point` to take `None` as a coordinate, which allows skipping this dimension in mesh generation. But I'm not sure what's the best way to visualization this. When one of the 2d coordinate is `None`, snapping points become "snapping line".  I plot a segment of line that doesn't cover the entire region, so that it doesn't get confused with grid lines. Otherwise, it's hard to tell if there is a grid line at the position of "snapping line".

Reply:
Here is the figure under current implementation where snapping point takes the form `(x, y, None)`
![image](https://github.com/user-attachments/assets/26d087a6-9d48-4664-92cc-a8cf06cc0a8b)
![image](https://github.com/user-attachments/assets/6b912d7c-4272-41ba-af4b-562ca878fc03)